artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/AmqpCoreConverter.java

```diff
@@ -324,7 +327,26 @@ private static ServerJMSMessage processProperties(ServerJMSMessage jms, Properti
             jms.setJMSType(properties.getSubject());
          }
          if (properties.getReplyTo() != null) {
-            jms.setJMSReplyTo(new ServerDestination(properties.getReplyTo()));
+            Byte value = (Byte)(annotations != null ? annotations.getValue().get(AMQPMessageSupport.JMS_REPLY_TO_TYPE_MSG_ANNOTATION) : AMQPMessageSupport.QUEUE_TYPE);
+
+
+            switch (value) {
+               case AMQPMessageSupport.QUEUE_TYPE:
+                  org.apache.activemq.artemis.reader.MessageUtil.setJMSReplyTo(jms.getInnerMessage(), ActiveMQDestination.QUEUE_QUALIFIED_PREFIX + properties.getReplyTo());
+                  break;
+               case AMQPMessageSupport.TEMP_QUEUE_TYPE:
+                  org.apache.activemq.artemis.reader.MessageUtil.setJMSReplyTo(jms.getInnerMessage(), ActiveMQDestination.TEMP_QUEUE_QUALIFED_PREFIX + properties.getReplyTo());
+                  break;
+               case AMQPMessageSupport.TOPIC_TYPE:
+                  org.apache.activemq.artemis.reader.MessageUtil.setJMSReplyTo(jms.getInnerMessage(), ActiveMQDestination.TOPIC_QUALIFIED_PREFIX + properties.getReplyTo());
+                  break;
+               case AMQPMessageSupport.TEMP_TOPIC_TYPE:
+                  org.apache.activemq.artemis.reader.MessageUtil.setJMSReplyTo(jms.getInnerMessage(), ActiveMQDestination.TEMP_TOPIC_QUALIFED_PREFIX + properties.getReplyTo());
+                  break;
+               default: // in theory this should never happen
```

Comment:
Why would this never happen in theory, AMQP messages can have message annotations and also use a reply to value and NOT be from a JMS type client.

Reply:
Agree here.

Reply:
Worth adding some non jms tests

Reply:
@tabish121 if the value is not set here, it will use QUEUE-TYPE, from line 330:

```java
            Byte value = (Byte)(annotations != null ? annotations.getValue().get(AMQPMessageSupport.JMS_REPLY_TO_TYPE_MSG_ANNOTATION) : AMQPMessageSupport.QUEUE_TYPE);
```

Reply:
I'm removing the note